### PR TITLE
*: refactor metamorphic constants

### PR DESCRIPTION
**Pull request by AlexTalks**
While previously metamorphic constants were part of `pkg/util`, this
creates new package `pkg/util/metamorphic` that can be used as a minimal
dependency for all tests that use code depending on metamorphic
variables. This list of tests can be queried using:
```
bazel query 'let testpkgs = kind(go_test, //pkg/...) in
  $testpkgs intersect allpaths($testpkgs, //pkg/util/metamorphic:metamorphic)'
```
Each of these packages should be updated with a `TestMain(..)` that logs
metamorphic constants at the start, to ensure that any tests using code
that depends on metamorphic variables will log their values.

The list of Go binary targets that use metamorphic variables can also be
queried with the following, although not all may be able to be built
with the `crdb_test` flag:
```
bazel query 'let binpkgs = kind(go_binary, //pkg/...) in
  $binpkgs intersect allpaths($binpkgs, //pkg/util/metamorphic:metamorphic)'
```

Epic: none

Release note: None

### CODE REVIEW

File: pkg/util/metamorphic/constants.go [hunk 1]
```diff
@@ -8,7 +8,7 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-package util
+package metamorphic
```
Comment (RaduBerinde) [inline pkg/util/metamorphic/constants.go:11]: Consider adding a package-level doc comment explaining at a high level what the package does / what metamorphic constants are.

**Review by RaduBerinde — APPROVED** [review]
LGTM.

Consider simplifying the API names now that the package has a specific name. E.g. `ConstantWithMetamorphicTestRange` could be `metamorphic.IntWithTestRange`, `ConstantWithMetamorphicTestBool` can be just `metamorphic.Bool`, etc.

**Review by rickystewart — APPROVED** [review]
LGTM.

I notice that the `skip` package depends on `metamorphic` via `skip.UnderMetamorphic[WithIssue]`, which causes every package that imports `skip` to import `pkg/util/metamorphic` indirectly. There are very few uses of `skip.UnderMetamorphic` so this can be easily addressed (as a follow-up).

**Comment by rickystewart** [comment]
Oh, I also think that Radu's advice to update the names of the API's are good too.

**Comment by RaduBerinde** [comment]
We could move `skip.UnderMetamorphic` to `metamorphic.SkipTest()` or similar.

**Pull request closed without merging**